runtime/src/main/cpp/Memory.cpp

```diff
@@ -1063,14 +1064,26 @@ void freeAggregatingFrozenContainer(ContainerHeader* container) {
   MEMORY_LOG("Freeing subcontainers done\n");
 }
 
+// Not inlining this call as it affects deallocation performance for
+// all types.
+NO_INLINE RUNTIME_NOTHROW void runFinalizers(ObjHeader* obj) {
+    auto* type_info = obj->type_info();
+    if (type_info == theCleanerImplTypeInfo) {
+        DisposeCleaner(obj);
```

Comment:
What exactly prevents `obj->clean` from being dead at this point?

Reply:
Fixed by putting the entire `clean` lambda inside stable pointer.